src/commands/importDocuments.ts

```diff
@@ -0,0 +1,127 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { NewDocument } from 'documentdb';
+import * as fse from 'fs-extra';
+import { DocDBDocumentsTreeItem } from 'src/docdb/tree/DocDBDocumentsTreeItem';
+import * as vscode from 'vscode';
+import { AzureTreeDataProvider, IAzureParentNode, parseError, UserCancelledError } from 'vscode-azureextensionui';
+import { DocDBCollectionTreeItem } from '../docdb/tree/DocDBCollectionTreeItem';
+import { ext } from '../extensionVariables';
+import { MongoCollectionTreeItem } from '../mongo/tree/MongoCollectionTreeItem';
+import { withProgress } from '../mongo/tree/MongoDatabaseTreeItem';
+
+export async function importDocuments(tree: AzureTreeDataProvider, uris: vscode.Uri[] | undefined, collectionNode: IAzureParentNode<MongoCollectionTreeItem | DocDBCollectionTreeItem> | undefined): Promise<void> {
+    if (!uris) {
+        uris = await askForDocuments();
+    }
+    let ignoredUris: vscode.Uri[] = []; //account for https://github.com/Microsoft/vscode/issues/59782
+    uris = uris.filter((uri) => {
+        if (uri.fsPath.endsWith('.json')) {
+            return true;
+        } else {
+            ignoredUris.push(uri);
+            return false;
+        }
+    });
+    if (ignoredUris.length) {
+        ext.outputChannel.appendLine(`The following selected files are not json: ${ignoredUris.map(uri => uri.fsPath).join(',')}. \nWe are ignoring these.`);
+        ext.outputChannel.show();
+    }
+    const documents = await withProgress(parseDocumentsForErrors(uris), "Parsing documents...", vscode.ProgressLocation.Notification);
```

Comment:
Parsing documents should happen after the user has gone through all UI. This lets them cancel before any long running notification and it avoids delays between input steps.

Specifically, this should be after `showNodePicker`